src/Core__Math.resi

```diff
@@ -0,0 +1,297 @@
+  Math.Int.clz32(4) // 29
+  ```
+  */
+  @val
+  external clz32: int => int = "Math.clz32"
+
+  /**
+  `imul(a, b)` returns 32-bit integer multiplication. Use this only when you
+  need to optimize performance of multiplication of numbers stored as 32-bit
+  integers. See [`Math.imul`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Math/imul) on MDN.
+
+  ## Examples
+
+  ```rescript
+  Math.Int.imul(3, 4) // 12
+  Math.Int.imul(-5, 12) // 60
+  ```
+  */
+  @val external imul: (int, int) => int = "Math.imul"
+
+  /**
+  `min(a, b)` returns the minimum of its two integer arguments. See [`Math.min`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Math/min) on MDN.
+
+  ## Examples
+
+  ```rescript
+  Math.Int.min(1, 2) // 1
+  Math.Int.min(-1, -2) // -2
+  ```
+  */
+  @val external min: (int, int) => int = "Math.min"
+
+  /**
+  `minMany(arr)` returns the minimum of the integers in the given array `arr`.
+  See [`Math.min`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Math/min) on MDN.
+
+  ## Examples
+
+  ```rescript
+  Math.Int.minMany([1, 2]) // 1
+  Math.Int.minMany([-1, -2]) // -2
+  Math.Int.minMany([]) == 0 // false
+  ```
+  */
+  @variadic @val external minMany: array<int> => int = "Math.min"
+
+  /**
+  `max(a, b)` returns the maximum of its two integer arguments. See [`Math.max`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Math/max) on MDN.
+
+  ## Examples
+
+  ```rescript
+  Math.Int.max(1, 2) // 2
+  Math.Int.max(-1, -2) // -1
+  ```
+  */
+  @val external max: (int, int) => int = "Math.max"
+
+  // NOTE: `Math.Int.maxMany([])` compile to `Math.max()` and return `Infinity`
+  // In JS `Math.max([])` return 0
+  // We should create a wrapper to handle an empty array?
+  // An Example: `Math.Int.maxMany([])` not return an int.
+  // Same problem with `Math.Int.minMany`
+
+  /**
+  `maxMany(arr)` returns the maximum of the integers in the given array `arr`.
+  See [`Math.max`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Math/max) on MDN.
+
+  ## Examples
+
+  ```rescript
+  Math.Int.maxMany([1, 2]) // 2
+  Math.Int.maxMany([-1, -2]) // -1
+  Math.Int.maxMany([]) == 0 // false
+  ```
+  */
+  @variadic @val external maxMany: array<int> => int = "Math.max"
+
+  // NOTE: the return type may be not int, example `pow(3, ~exp=-3)` return a float
+  // We should remove this function?
+
+  /**
+  `pow(a, ~exp)` raises the given base `a` to the given exponent `exp`. See [`Math.pow`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Math/pow) on MDN.
+
+  ## Examples
+
+  ```rescript
+  Math.Int.pow(2, ~exp=4) // 16
+  Math.Int.pow(3, ~exp=4) // 81
+  ```
+  */
+  @val external pow: (int, ~exp: int) => int = "Math.pow"
+
+  // NOTE: In JS `Math.sin(2)` return a float
+  @val external sign: int => int = "Math.sign"
```

Comment:
`sign` function signature should be `float => float`, since `Math.sign` return a float?

[MDN](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Math/sin)